From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 17 Jun 2018 21:27:00 +0200
Subject: [PATCH] {MAYBE UPS} scripts/unifdef: fix gcc 8 warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 8 reports:

    scripts/unifdef.c: In function ‘Mpass’:
    scripts/unifdef.c:453:28: error: ‘strncpy’ output truncated before
    terminating nul copying 4 bytes from a string of the same length
    [-Werror=stringop-truncation]
     static void Mpass (void) { strncpy(keyword, "if  ", 4); Pelif(); }
                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
---
 scripts/unifdef.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/unifdef.c b/scripts/unifdef.c
index fba8ff17d8c6..ea644ace4a0d 100644
--- a/scripts/unifdef.c
+++ b/scripts/unifdef.c
@@ -450,7 +450,7 @@ static void Idrop (void) { Fdrop();  ignoreon(); }
 static void Itrue (void) { Ftrue();  ignoreon(); }
 static void Ifalse(void) { Ffalse(); ignoreon(); }
 /* modify this line */
-static void Mpass (void) { strncpy(keyword, "if  ", 4); Pelif(); }
+static void Mpass (void) { strncpy(keyword, "if  ", 5); Pelif(); }
 static void Mtrue (void) { keywordedit("else");  state(IS_TRUE_MIDDLE); }
 static void Melif (void) { keywordedit("endif"); state(IS_FALSE_TRAILER); }
 static void Melse (void) { keywordedit("endif"); state(IS_FALSE_ELSE); }
-- 
